rnd/autogpt_libs/autogpt_libs/auth/decorator.py

```diff
@@ -0,0 +1,49 @@
+import functools
+import logging
+import typing
+
+from .middleware import auth_middleware
+import fastapi
+
+logger = logging.getLogger("uvicorn.error")
+
+
+async def get_user(
+    payload: dict = fastapi.Depends(auth_middleware),
+) -> dict[str, str]:
+    if not payload:
+        # This handles the case when authentication is disabled
+        user_id = "3e53486c-cf57-477e-ba2a-cb02dc828e1a"
+
+    user_id = payload.get("sub")
+    logger.info(payload["role"])
+    if not user_id:
+        raise fastapi.HTTPException(
+            status_code=401, detail="User ID not found in token"
+        )
+
+    return {"user_id": user_id, "role": payload["role"]}
+
+
+def require_auth(admin_only: bool = False) -> typing.Callable:
```

Comment:
I'm not sure creating a decorator was wise @ntindle 

It does not seem to be in line with the fastapi docs